monai/losses/dice.py

```diff
@@ -78,36 +83,39 @@ def forward(self, pred, ground, smooth=1e-5):
                 warnings.warn("single channel prediction, `include_background=False` ignored.")
         else:
             if self.do_softmax:
-                pred = torch.softmax(pred, 1)
+                input = torch.softmax(input, 1)
             if self.to_onehot_y:
-                ground = one_hot(ground, n_pred_ch)
+                target = one_hot(target, n_pred_ch)
             if not self.include_background:
                 # if skipping background, removing first channel
-                ground = ground[:, 1:]
-                pred = pred[:, 1:]
-                assert ground.shape == pred.shape, "ground truth one-hot has differing shape (%r) from pred (%r)" % (
-                    ground.shape,
-                    pred.shape,
-                )
+                target = target[:, 1:]
+                input = input[:, 1:]
+        assert (
+            target.shape == input.shape
+        ), f"ground truth has differing shape ({target.shape}) from input ({input.shape})"
 
         # reducing only spatial dimensions (not batch nor channels)
-        reduce_axis = list(range(2, len(pred.shape)))
-        intersection = torch.sum(ground * pred, reduce_axis)
+        reduce_axis = list(range(2, len(input.shape)))
+        intersection = torch.sum(target * input, reduce_axis)
 
         if self.squared_pred:
-            ground = torch.pow(ground, 2)
-            pred = torch.pow(pred, 2)
+            target = torch.pow(target, 2)
+            input = torch.pow(input, 2)
 
-        ground_o = torch.sum(ground, reduce_axis)
-        pred_o = torch.sum(pred, reduce_axis)
+        ground_o = torch.sum(target, reduce_axis)
+        pred_o = torch.sum(input, reduce_axis)
 
         denominator = ground_o + pred_o
 
         if self.jaccard:
             denominator -= intersection
 
-        f = (2.0 * intersection + smooth) / (denominator + smooth)
-        return 1.0 - f.mean()  # final reduce_mean across batches and channels
+        f = 1.0 - (2.0 * intersection + smooth) / (denominator + smooth)
+        if self.reduction == "sum":
```

Comment:
Suggest to simply this `reduction` logic according to PyTorch [source code](https://github.com/pytorch/pytorch/blob/master/torch/nn/functional.py#L2613) as all loss functions need it:
```py
if reduction != 'none':
    ret = torch.mean(ret) if reduction == 'mean' else torch.sum(ret)
```
Thanks.

Reply:
thanks, there are different styles in the link you provided, now following the robust one https://github.com/pytorch/pytorch/blob/master/torch/nn/functional.py#L2756

Reply:
Looks good.
Thanks.